src/Server/Repositories/ClaraJobRepository.cs

```diff
@@ -0,0 +1,236 @@
+using System.IO;
+using System.IO.Abstractions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Nvidia.Clara.DicomAdapter.Server.Repositories
+{
+    public class ClaraJobRepository : IJobRepository
+    {
+        private const int MaxRetryLimit = 3;
+
+        private readonly IOptions<DicomAdapterConfiguration> _configuration;
+        private readonly ILogger<ClaraJobRepository> _logger;
+        private readonly IFileSystem _fileSystem;
+        private readonly IDicomAdapterRepository<InferenceJob> _inferenceJobRepository;
+
+        public ServiceStatus Status { get; set; } = ServiceStatus.Unknown;
+
+        public ClaraJobRepository(
+            ILogger<ClaraJobRepository> logger,
+            IOptions<DicomAdapterConfiguration> configuration,
+            IFileSystem fileSystem,
+            IDicomAdapterRepository<InferenceJob> inferenceJobRepository)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+            _fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+            _inferenceJobRepository = inferenceJobRepository ?? throw new ArgumentNullException(nameof(inferenceJobRepository));
+        }
+
+        public async Task Add(Job job, string jobName, IList<InstanceStorageInfo> instances)
+        {
+            Guard.Against.Null(job, nameof(job));
+            Guard.Against.Null(jobName, nameof(jobName));
+            Guard.Against.NullOrEmpty(instances, nameof(instances));
+
+            using (_logger.BeginScope(new LogginDataDictionary<string, object> { { "JobId", job.JobId }, { "PayloadId", job.PayloadId } }))
+            {
+                var inferenceJob = CreateInferenceJob(job, jobName, instances);
+
+                // Makes a copy of the payload to support multiple pipelines per AE Title.
+                // Future, consider use of persisted payloads.
+                MakeACopyOfPayload(inferenceJob);
+
+                await Policy
+                    .Handle<Exception>()
+                    .WaitAndRetryAsync(
+                        3,
+                        retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt)),
+                        (exception, timeSpan, retryCount, context) =>
+                    {
+                        _logger.Log(LogLevel.Error, exception, $"Error saving inference request. Waiting {timeSpan} before next retry. Retry attempt {retryCount}.");
+                    })
+                    .ExecuteAsync(async () =>
+                    {
+                        await _inferenceJobRepository.AddAsync(inferenceJob);
+                        await _inferenceJobRepository.SaveChangesAsync();
+                    })
+                    .ConfigureAwait(false);
+            }
+        }
+
+        public async Task Update(InferenceJob inferenceJob, InferenceJobStatus status)
```

Comment:
ensure `inferenceJob` and `status` before first use?

Reply:
Fixed 🙏